keep client and controller in a single image

phoracek opened the pull request:
With this PR we:
- move client to the same image as controller since they share all dependencies
- use multi-stage builds so it is easier to automate building
- set images repo to nmstate quay.io registry

phoracek commented:
@SchSeba could you please take a look? There are some supporting patches not really part of the PR topic. I just wanted some tweaks to piggy back on the main commit. It might be easier to review this commit by commit. Thanks!

yuvalif commented on Makefile:
```diff
@@ -1,51 +1,39 @@
-IMAGE_REGISTRY ?= yuvalif
+build:
+	cd cmd/state-handler && go fmt && go vet && go build
+	cd cmd/policy-handlerm && go fmt && go vet && go build
```
why "policy-handlerm"?

phoracek replied:
Fixed

yuvalif commented on cmd/state-handler/main.go:
```diff
@@ -0,0 +1,126 @@
+}
+
+func client() {
+	flag.Parse()
```
why do you need to call "parse" again?

phoracek replied:
Fixed

yuvalif commented on cmd/state-handler/main.go:
```diff
@@ -0,0 +1,126 @@
+	nmstateClient, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatalf("Error building nmstate clientset: %s", err.Error())
+	}
```
minor: BuildConfigFromFlags(), clientset.NewForConfig(), GetNsmespace() and GetHostName() are common to both client and controller, could be taken out of the specific functions.

yuvalif commented on cmd/state-handler/main.go:
```diff
@@ -0,0 +1,126 @@
+		klog.Fatalf("Failed to get host name\n")
+	}
+
+	for _, state := range list.Items {
```
inside the controller code, don't we have similar initialization to the one we perform here?
or is it all based on getting updated from k8s?

phoracek replied:
oh, missed this one. will fix it in a second

phoracek replied:
I don't see how could we merge the logic at this point. I need to research more about the controller. Could we take this PR as it is?

phoracek commented:
@yuvalif thanks for the reviews. I addressed all of your comments. Most of them are resolved inhttps://github.com/nmstate/kubernetes-nmstate/pull/27/commits/682f6df898f87842ebcc5986330b2699ae8def0a
